src/SourceBuild/content/test/tests.proj

```diff
@@ -7,4 +7,49 @@
                       Condition="'$(ShortStack)' != 'true' and '$(PortableBuild)' == 'true' and '$(PgoInstrument)' != 'true'" />
   </ItemGroup>
 
-</Project>
\ No newline at end of file
+  <Target Name="RunCoreTests">
+    <MSBuild Projects="$(MSBuildThisFileFullPath)"
+             Properties="Configuration=$(Configuration)"
+             ContinueOnError="true"
+             Targets="Build;VSTest" />
+
+    <PropertyGroup>
+      <CoreTestsResult>$(MSBuildLastTaskResult)</CoreTestsResult>
+    </PropertyGroup>
+  </Target>
+
+  <!-- Scenarios tests are handled separately from the other test projects. This is because the output of the scenario tests is
+       an executable that is used to run the tests and so using the VSTest runner doesn't work with them. They also require
+       some setup steps to be configured with the VMR. -->
+  <Target Name="RunScenarioTests">
+    <PropertyGroup>
+      <_HostArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_HostArchitecture>
+      <_RunScenarioTests>true</_RunScenarioTests>
+
+      <!-- Skip scenario tests if the host architecture is different from the target architecture since the tests
+           require the ability to execute the built SDK. But the CLI is not capable of running on a host with a 
+           different architecture (i.e. "cannot execute binary file: Exec format error"). -->
+      <_RunScenarioTests Condition="'$(_HostArchitecture.ToLower())' != '$(TargetArchitecture.ToLower())'">false</_RunScenarioTests>
```

Comment:
Can't this be expressed via the already defined variables in Directory.Build.props?

I.e.
```xml
<_RunScenarioTests Condition="'$(BuildArchitecture)' != '$(TargetArchitecture)''">false</_RunScenarioTests>
```

Reply:
Fixed in https://github.com/dotnet/installer/pull/19222/commits/b7bf1bb33102e085eac8e96121d081e3c288a32b